internal/datastore/common/sql.go

```diff
@@ -104,6 +104,59 @@ func (sqf SchemaQueryFilterer) FilterToRelation(relation string) SchemaQueryFilt
 	return sqf
 }
 
+// FilterWithSubjectsFilter returns a new SchemaQueryFilterer that is limited to resources with
+// subjects that match the specified filter.
+func (sqf SchemaQueryFilterer) FilterWithSubjectsFilter(filter datastore.SubjectsFilter) SchemaQueryFilterer {
```

Comment:
Add a unit test for this method.

Reply:
Where do we have the unit tests for these methods defined? I don't see one for `FilterToSubjectFilter` either

Reply:
I don't know if you're just being cheeky, but we don't have any unit tests for these helper sql methods yet. We should add them when they're purely functional though. They would go in `sql_test.go`.

Reply:
No, I was just making sure I didn't miss them. I'll add a bunch.

Reply:
Added